File: Byte&Buy.Desktop/src/ByteBuy.Services/DTO/Rental/RentalLenderResponse.cs
using ByteBuy.Services.DTO.Image;
using ByteBuy.Services.DTO.Money;
using ByteBuy.Services.DTO.Rental.Enum;

namespace ByteBuy.Services.DTO.Rental;

public sealed record RentalLenderResponse(
    Guid Id,
    RentalStatus Status,
    string ItemName,
    int Quantity,
    MoneyDto TotalPricePaid,
    ImageThumbnailDto Thumbnail,
    DateTime DateCreated,
    int RentalDays,
    string BorrowerName,
    string BorrowerEmail,
    DateTime StartingRentalDate,
    DateTime EndingRentalDate);

File: Byte&Buy.AngularSPA/src/app/core/dto/rental/rental-lender-response.ts
import { MoneyDto } from "../common/money-dto";
import { ImageThumbnailDto } from "../image/image-thumbnail";
import { RentalStatus } from "./enum/rental-status";

export interface RentalLenderResponse {
  id: string;
  status: RentalStatus;
  itemName: string;
  quantity: number;
  totalPricePaid: MoneyDto;
  thumbnail: ImageThumbnailDto;
  dateCreated: string;
  rentalDays: number;
  borrowerName: string;
  borrowerEmail: string;
  startingRentalDate: string; 
  endingRentalDate: Date;
}